resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -47,6 +49,16 @@ private[spark] class ExecutorPodsAllocator(
 
   private val podCreationTimeout = math.max(podAllocationDelay * 5, 60000)
 
+  // maxNumExecutorFailures must be less than Int.MaxValue
+  private lazy val maxNumExecutorFailures = conf.get(KUBERNETES_MAX_EXECUTOR_FAILURES)
```

Comment:
Is this safe to use `val`? Although this is `lazy val`, the underlying `totalExpectedExecutors` is used in two ways.
```
podAllocator.setTotalExpectedExecutors(initialExecutors)
```

```
podAllocator.setTotalExpectedExecutors(requestedTotal)
```

This looks a little different from YARN.

Reply:
It works fine without executor dynamicAllocation, but if we need dynamicAllocation, then this should be def.